### PR TITLE
Only use node types where necessary

### LINKED ISSUE #8343: @storybook/angular & @storybook/addons require node types
Can't find it in the source code, but the published `@storybook/angular` & `@storybook/addons` version 5.2.3 types require node types:

image
image

That conflicts with client code, e.g. we see lots of:

```log
Error:(10, 9) TS2322: Type 'Timeout' is not assignable to type 'number'.
```

Because the node.js setTimeout definition returns `Timeout` and the dom implementation returns `number`. Is that reference to node.js really required? Given that much of the code also runs in the browser, node.js types are kind of misleading anyway.

### PR DESCRIPTION
**Pull request by aaronjensen**
Fixes #8343.

## What I did

Removed node from root tsconfig and added it where necessary.

## How to test

- Is this testable with Jest or Chromatic screenshots?
No
- Does this need a new example in the kitchen sink apps?
No
- Does this need an update to the documentation?
No

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by aaronjensen** [comment]
I don't know if I should be concerned by the above failures... if so, let me know and please give me a hint as to what they mean. Thanks!

**Comment by ndelangen** [comment]
The now deployment is a bit concerning, but I can't debug it, I'll have to run locally to debug.

File: lib/addons/tsconfig.json [hunk 1]
```diff
@@ -1,7 +1,8 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "rootDir": "./src"
```
Comment (ndelangen) [inline lib/addons/tsconfig.json]: I think this line is causing problems for some addons (addon-docs) in particular it seems:

```
lerna ERR! yarn run prepare exited 2 in '@storybook/addon-docs'
lerna ERR! yarn run prepare stdout:
$ node ../../scripts/prepare.js
src/preset.ts(50,56): error TS2345: Argument of type '(framework: string) => string | number' is not assignable to parameter of type 'FrameworkPresetMapper'.
  Type 'string | number' is not assignable to type 'string | void'.
    Type 'number' is not assignable to type 'string | void'.
```

Comment (ndelangen) [reply lib/addons/tsconfig.json]: found a fix

**Review by ndelangen — APPROVED** [review]

**Comment by aaronjensen** [comment]
Thanks for the fix & the merge. I was hoping it could go into master and be a point release, but we can wait for 6.0.

**Comment by ndelangen** [comment]
@shilman added it to the 6.0.0 milestone, So I'm going with that.

We can ask him if it's ok to be patched into the 5.3.x release?

**Comment by shilman** [comment]
I'm fine to patch after it's been out in a 6.0 alpha for a week or two.

**Comment by ndelangen** [comment]
sounds good to me.

**Comment by aaronjensen** [comment]
This has been out for a week or two, if it hasn't made it to 5.x is it okay for it to go there now? Thanks!

**Comment by ndelangen** [comment]
@aaronjensen 5.3 will be released in a few days, then we'll start publishing alpha's of 6.0.0.

@shilman said that if this was in 6.0.0 alpha for a bit he's consider patching this change to 5.3.x

**Comment by aaronjensen** [comment]
Ah, thank you for the clarification.

**Pull request merged**